File: Assets/Tests/Resources/performance.mjs
// cs.PerformanceHelper.ReturnNumber(3);
// const start = Date.now();
// for (let i = 0; i < 1000000; i++) 
// {
//     cs.PerformanceHelper.ReturnNumber(3);
// }
// // cs.UnityEngine.Debug.Log('Puerts Number:');
// cs.PerformanceHelper.JSNumber.text = 'Puerts Number:' + (Date.now() - start) + 'ms'

cs.PerformanceHelper.ReturnVector(1, 2, 3)
const start2 = Date.now();
for (let i = 0; i < 1000000; i++) 
{
    cs.PerformanceHelper.ReturnVector(1, 2, 3);
}
// cs.UnityEngine.Debug.Log('Puerts Vector:');
// cs.UnityEngine.Debug.Log((Date.now() - start2));
cs.PerformanceHelper.JSVector.text = 'Puerts Vector:' + (Date.now() - start2) + 'ms'
// const fibcache = [0, 1]
// function fibonacci(level) {
//     if (level == 0) return 0;
//     if (level == 1) return 1;
//     return fibonacci(level - 1) + fibonacci(level - 2);
// }

// const start3 = Date.now();
// for (let i = 0; i < 100000; i++) {
//     fibonacci(12)
// }
// // cs.UnityEngine.Debug.Log('Puerts fibonacci:');
// // cs.UnityEngine.Debug.Log((Date.now() - start3));
// cs.PerformanceHelper.JSFibonacci.text = 'Puerts fibonacci:' + (Date.now() - start3) + 'ms'

File: Assets/Tests/Performance.cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.UI;

[XLua.LuaCallCSharp]
public class PerformanceHelper
{
    public static int ReturnNumber(int num)
    {
        return num;
    }
    public static Vector3 ReturnVector(int x, int y, int z)
    {
        return new Vector3(x, y, z);
    }

    public static Text JSNumber;
    public static Text JSVector;
    public static Text JSFibonacci;
    public static Text LuaNumber;
    public static Text LuaVector;
    public static Text LuaFibonacci;
}

public class Performance : MonoBehaviour
{
    public Text JSNumber;
    public Text JSVector;
    public Text JSFibonacci;
    public Text LuaNumber;
    public Text LuaVector;
    public Text LuaFibonacci;
    // Start is called before the first frame update
    void Start()
    {
        PerformanceHelper.JSNumber = JSNumber;
        PerformanceHelper.JSVector = JSVector;
        PerformanceHelper.JSFibonacci = JSFibonacci;
        PerformanceHelper.LuaNumber = LuaNumber;
        PerformanceHelper.LuaVector = LuaVector;
        PerformanceHelper.LuaFibonacci = LuaFibonacci;

        Puerts.JsEnv JsEnv = new Puerts.JsEnv();
        JsEnv.ExecuteModule("performance.mjs");
        
        // XLua.LuaEnv env = new XLua.LuaEnv();
        // env.DoString(Resources.Load<TextAsset>("performance.lua").text);
    }

    // Update is called once per frame
    void Update()
    {
        
    }
}
